### PR TITLE
allow to pass embed to getCurrentUser

### PR DESCRIPTION
**Pull request by Amtron**

### CODE REVIEW

File: typescript/src/merchi.ts [hunk 1]
```diff
@@ -229,12 +229,12 @@ export class Merchi {
     return apiFetch(resource, options);
   };
 
-  public getCurrentUser = () => {
+  public getCurrentUser = (embed?: any) => {
```
Comment (DanielCollins) [inline typescript/src/merchi.ts]: embed must be an embed descriptor, not `any`.

File: typescript/src/merchi.ts [hunk 2]
```diff
@@ -229,12 +229,12 @@ export class Merchi {
...
     if (!this.sessionToken) {
       return Promise.resolve(null);
     }
     return this.Session.get(this.sessionToken, {
-      embed: { user: { enrolledDomains: {domain: {}} } }
+      embed: embed ? embed : { user: { enrolledDomains: {domain: {}} } }
```
Comment (DanielCollins) [inline typescript/src/merchi.ts]: this breaks the line below in the case that the caller provides an embed descriptor not including `user`.

the provided embed descriptor should be merged with the minimum requirement, it should not replace the minimum requirement

Comment (Amtron) [reply typescript/src/merchi.ts]: I would have thought to do that. I would have thought that if I provided an embed WYSIWYG. But I like your idea of having a minimum embed.

File: typescript/src/merchi.ts (hunk 1 above)
Comment (Amtron) [reply typescript/src/merchi.ts]: yep.

**Pull request merged**